android/src/main/kotlin/com/rightbite/denisr/KlaviyoFlutterPlugin.kt

```diff
@@ -72,21 +74,42 @@ class KlaviyoFlutterPlugin : MethodCallHandler, FlutterPlugin {
             }
 
             METHOD_UPDATE_PROFILE -> {
-                call.arguments<HashMap<String, String>>()?.let { profileMap ->
-                    val profile = Profile(
-                        profileMap.map { (key, value) ->
-                            ProfileKey.CUSTOM(key) to value
-                        }.toMap()
-                    )
-
-                    Klaviyo.setProfile(profile)
-                    Log.d(
-                        TAG,
-                        "Profile updated: ${Klaviyo.getExternalId()}, profileMap: $profileMap"
-                    )
+                try {
+                val profilePropertiesRaw = call.arguments<Map<String, Any>?>()
+                    ?: throw RuntimeException("Profile properties not exist")
+                
+                val customPropertiesKey = "properties"
```

Comment:
[nit] Better move this const out of class to the top of file:
`private const val PROFILE_PROPERTIES_KEY = "properties"`